components/sup/src/manager/service/mod.rs

```diff
@@ -485,11 +487,13 @@ impl Service {
 
     fn post_run(&mut self) {
         if let Some(ref hook) = self.hooks.post_run {
-            hook.run(
+            self.post_run = hook.run(
                 &self.service_group,
                 &self.pkg,
                 self.svc_encrypted_password.as_ref(),
-            );
+            ).0 == 0;
```

Comment:
I'd prefer adding functions to the `ExitCode` struct that allow us to ask if it was successful or not, instead of doing direct (and relatively opaque) manipulations like this.

It would also let you just store the post-run hook result directly, instead of keeping a boolean, which could yield more expressive code elsewhere.